cmd/gcp-controller-manager/node_csr_approver.go

```diff
@@ -860,9 +893,29 @@ func shouldDeleteNode(ctx *controllerContext, node *v1.Node, getInstance func(*c
 	}
 	// Instance with no alias range is route based, for which node object deletion is unnecessary.
 	klog.V(2).Infof("Instance %q has no alias range.", inst.Name)
+
 	return false, nil
 }
 
+func deleteAllPodsBoundToNode(ctx *controllerContext, nodeName string) error {
+	klog.V(1).Infof("start delete all pods pod on node %v", nodeName)
+	errs := []error{}
+
+	podList, err := ctx.client.CoreV1().Pods(metav1.NamespaceAll).List(context.TODO(), metav1.ListOptions{FieldSelector: "spec.nodeName=" + nodeName})
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	for _, pod := range podList.Items {
+		err := ctx.client.CoreV1().Pods(pod.Namespace).Delete(context.TODO(), pod.Name, *metav1.NewDeleteOptions(0))
+		if err != nil {
+			errs = append(errs, err)
+		}
+	}
+	klog.Infof("end delete all pods pod on node %v", nodeName)
```

Comment:
ditto let's remove this.

Maybe instead let's add something like the following (only log if we did in fact delete pods). How about we declare a slice of pod names and append each pod name we delete to slice if there is no error returned. Then at the end if the slice is not empty (we deleted some pods), we log the node name and all the pods deleted.